src/m365/spfx/commands/project/project-github-workflow-add.spec.ts

```diff
@@ -149,12 +154,74 @@ describe(commands.PROJECT_GITHUB_WORKFLOW_ADD, () => {
       return '';
     });
 
+    sinon.stub(command as any, 'getProjectVersion').returns('1.21.1');
+
     const writeFileSyncStub: sinon.SinonStub = sinon.stub(fs, 'writeFileSync').resolves({});
 
     await command.action(logger, { options: { name: 'test', branchName: 'dev', manuallyTrigger: true, skipFeatureDeployment: true, loginMethod: 'user', scope: 'sitecollection' } } as any);
     assert(writeFileSyncStub.calledWith(path.join(process.cwd(), projectPath, '/.github', 'workflows', 'deploy-spfx-solution.yml')), 'workflow file not created');
   });
 
+  it('handles error with unknown version of SPFx', async () => {
+    sinon.stub(command as any, 'getProjectRoot').returns(path.join(process.cwd(), projectPath));
+
+    sinon.stub(fs, 'readFileSync').callsFake((path, options) => {
+      if (path.toString().endsWith('package.json') && options === 'utf-8') {
+        return '{"name": "test"}';
+      }
+
+      return '';
+    });
+
+    sinon.stub(fs, 'existsSync').callsFake((fakePath) => {
+      if (fakePath.toString().endsWith('.github')) {
+        return true;
+      }
+      else if (fakePath.toString().endsWith('workflows')) {
+        return true;
+      }
+
+      return false;
+    });
+
+    sinon.stub(command as any, 'getProjectVersion').returns(undefined);
+
+    sinon.stub(fs, 'writeFileSync').callsFake(() => { throw 'error'; });
+
+    await assert.rejects(command.action(logger, { options: {} } as any),
+      new CommandError(`Unable to determine the version of the current SharePoint Framework project`, undefined));
+  });
+
+  it('handles error with not found node version', async () => {
+    sinon.stub(command as any, 'getProjectRoot').returns(path.join(process.cwd(), projectPath));
+
+    sinon.stub(fs, 'readFileSync').callsFake((path, options) => {
+      if (path.toString().endsWith('package.json') && options === 'utf-8') {
+        return '{"name": "test"}';
+      }
+
+      return '';
+    });
+
+    sinon.stub(fs, 'existsSync').callsFake((fakePath) => {
+      if (fakePath.toString().endsWith('.github')) {
+        return true;
+      }
+      else if (fakePath.toString().endsWith('workflows')) {
+        return true;
+      }
+
+      return false;
+    });
+
+    sinon.stub(command as any, 'getProjectVersion').returns('99.99.99');
+
+    sinon.stub(fs, 'writeFileSync').callsFake(() => { throw 'error'; });
```

Comment:
Is this `fs.writeFileSync` also throwing `'error'`, in real life? It's important to make the error message structure as realistic as possible.